trainlib/trainer.py

```diff
@@ -296,9 +296,17 @@ def _prepare_dirs(self) -> None:
         """Set up directories for saving logs, outputs and configs of current training session"""
         # create run_id, copy config file for reproducibility
         run_id = Path(self.config.run_id)
-        run_id.mkdir(exist_ok=True)
+        run_id.mkdir(exist_ok=True, parents=True)
         with open(run_id / "config.yaml", "w+") as f:
-            f.write(yaml.safe_dump(self.config))
+            config = dict(self.config)
+            # convert pathlib.Path to string, because of incompatibility with PyYAML
+            for path in ["run_id", "out_dir", "model_dir", "log_dir"]:
+                config[path] = str(config[path])
+
+            for path in ["data_dir", "train_csv", "valid_csv", "test_csv"]:
+                config["data"][path] = str(config["data"][path])
+
+            f.write(yaml.safe_dump(config))
```

Comment:
Suggestion for readability of the yaml

```suggestion
            f.write(yaml.safe_dump(config, indent=4))
```